src/Sulu/Component/Rest/ListBuilder/Expression/Doctrine/AbstractDoctrineExpression.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Rest\ListBuilder\Expression\Doctrine;
+
+use Doctrine\ORM\QueryBuilder;
+use Sulu\Component\Rest\ListBuilder\Expression\ExpressionInterface;
+
+/**
+ * Abstract definition for expressions used by the DoctrineListbuilder.
+ */
+abstract class AbstractDoctrineExpression implements ExpressionInterface
+{
+    /**
+     * Returns a statement for an expression.
+     *
+     * @param QueryBuilder $queryBuilder
+     *
+     * @return string
+     */
+    abstract public function getStatement(QueryBuilder $queryBuilder);
```

Comment:
Actually it would be quite easy to pass the query builder via the constructor, since you are using the `createWhereExpression` and other methods already anyway. Then this `getStatement` would have an empty parameter list. And I would think about a different name, since this method is doing more than just returning a statement...